### PR TITLE
add requestAnimationFrame and importNode

### PR DESCRIPTION
**Pull request by shutej**
In particular, advice on wrapping the [DOMHighResTimeStamp](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) would be helpful?

### CODE REVIEW

File: dom.go [hunk 1]
```diff
@@ -1080,16 +1087,21 @@ func (w *window) Stop() {
 
 // TODO reuse util.EventTarget
 
-func (w *window) AddEventListener(typ string, useCapture bool, listener func(Event)) func(o *js.Object) {
```
Comment (dominikh) [inline dom.go]: Please don't move AddEventListener around, it makes the diff noisier than it needs to be. It was fine being above the RemoveEventListener function.

File: dom.go [hunk 2]
```diff
@@ -1080,16 +1087,21 @@ func (w *window) Stop() {
...
-	wrapper := func(o *js.Object) { listener(wrapEvent(o)) }
-	w.Call("addEventListener", typ, wrapper, useCapture)
-	return wrapper
+func (w *window) RequestAnimationFrame(callback func()) func() {
+	w.Call("requestAnimationFrame", callback)
```
Comment (dominikh) [inline dom.go]: requestAnimationFrame returns an integer, which we should return as well. There's no point in returning the callback.

Comment (dominikh) [inline dom.go]: Also, please add support for cancelAnimationFrame. Yes, it's experimental, but I think it's safe to assume it'll stick around.

**Comment by dominikh** [comment]
I don't think we should add DOMHighResTimeStamp at all. Instead, we should use Go's time.Time -- RequestAnimationFrame should wrap the callback in its own function, that accepts the DOMHighResTimeStamp (as a *js.Object), converts it to a time.Time, and calls the real callback with it.

AddEventListener is a good example of the idea. It takes a *js.Object and turns it into an Event, we'd do the same just for *js.Object -> time.Time

File: dom.go [hunk 3]
```diff
@@ -691,6 +692,10 @@ func (d document) AdoptNode(node Node) Node {
 	return wrapNode(d.Call("adoptNode", node.Underlying()))
 }
 
+func (d document) ImportNode(node Node, deep bool) Node {
```
Comment (dominikh) [inline dom.go]: Split this PR into (at least) two commits please. ImportNode and RequestAnimationFrame are unrelated. I'm fine with them being in the same PR, as I'm going to accept both, but they should be independent commits.

File: dom.go (hunk 1 above)
Comment (shutej) [inline dom.go]: whoops, hacked the wrong copy.  fixed

File: dom.go (hunk 2 above)
Comment (shutej) [inline dom.go]: i think i did this correctly, ptal

**Comment by shutej** [comment]
I think I got this right?  PTAL

**Comment by dominikh** [comment]
Sorry, I didn't realise that the timestamp was relative to the start of the application, not an absolute time. We want to use time.Duration instead of time.Time then. Apart from that, your changes look good.

**Comment by shutej** [comment]
(I think this fixes duration?)

**Comment by dominikh** [comment]
Merged as bbbde021f90d13a4ccb948794bf8319b7b5aa337 - thank you..
